src/EloquentQueryModifier.php

```diff
@@ -355,7 +354,7 @@ protected function aggregateQuery(array $aggregate, array $columns, array $valid
         $function        = strtolower(key($aggregate));
 
         if (in_array($function, $allowed_aggregations, true) && in_array($column, $allowed_columns, true)) {
-            $this->query()->addSelect(DB::raw($function . '(' . $column . ') as aggregate'));
+            $this->query()->getQuery()->aggregate = ['columns' => [$column], 'function' => $function];
```

Comment:
Great catch! I like this approach much better. Didn't realize `DB::raw(...)` was vendor specific!